src/main/java/net/minestom/server/instance/block/BlockImpl.java

```diff
@@ -164,6 +164,7 @@ static Collection<Block> values() {
     public @Unmodifiable @NotNull Map<String, String> properties() {
         final PropertyType[] propertyTypes = PROPERTIES_TYPE.get(id());
         assert propertyTypes != null;
+        if (propertyTypes.length == 0) return Collections.emptyMap();
         final int length = propertyTypes.length;
```

Comment:
Should be `Map.of`. `Collections` util methods are generally unnecessary.

You can also move it below to get access to the `length` variable.

Reply:
The implementation, for your viewing pleasure
```java
static <K, V> Map<K, V> of() {
    return (Map<K,V>) ImmutableCollections.EMPTY_MAP;
}
```

Reply:
Ah should've checked, for some reason assumed it'd use varargs even though that class has overloads. Will push those two fixes.